### PR TITLE
#62 added filtering by rows and columns

### PR DESCRIPTION
**Pull request by klangner**
Added to the orc-exporter
* filter by rows
* filter by columns
* skip unsupported columns (e.g binary for csv)

### CODE REVIEW

File: src/bin/orc-export.rs [hunk 1]
```diff
@@ -13,30 +14,119 @@ struct Cli {
...
+        .enumerate()
+        .filter(|(_, nc)| match nc.data_type().to_arrow_data_type() {
+            DataType::Binary => true,
+            DataType::Decimal128(_, _) | DataType::Decimal256(_, _) => is_json,
+            _ => {
+                if let Some(cols) = &cli.columns {
+                    !cols.iter().any(|c| nc.name().eq(c))
+                } else {
+                    false
+                }
```
Comment (Jefffrey) [inline src/bin/orc-export.rs:99]: Can try push this to use the projection logic provided by ArrowReaderBuilder

File: src/bin/orc-export.rs [hunk 2]
```diff
@@ -13,30 +14,119 @@ struct Cli {
...
+    };
+
+    // Select columns which should be exported (Binary and Decimal are not supported)
+    let is_json = cli.format == Some(FileFormat::Json);
+    let skip_cols: Vec<usize> = metadata
+        .root_data_type()
+        .children()
+        .iter()
+        .enumerate()
+        .filter(|(_, nc)| match nc.data_type().to_arrow_data_type() {
```
Comment (Jefffrey) [inline src/bin/orc-export.rs:91]:
```suggestion
        // TODO: handle nested types
        .filter(|(_, nc)| match nc.data_type().to_arrow_data_type() {
```

File: src/bin/orc-export.rs [hunk 3]
```diff
@@ -13,30 +14,119 @@ struct Cli {
...
     #[arg(short, long)]
-    output: Option<PathBuf>,
-    // TODO: head=N
-    // TODO: convert_dates
-    // TODO: format=[csv|json]
-    // TODO: columns="col1,col2"
+    output_file: Option<PathBuf>,
+    /// Output format. If not provided then the output is csv
+    #[arg(value_enum, short, long)]
+    format: Option<FileFormat>,
```
Comment (Jefffrey) [inline src/bin/orc-export.rs:20]: I think clap provides a way to set a default, so wouldn't need to use option here :thinking:

**Review by Jefffrey — COMMENTED** [review]
Thanks for this, left some comments :+1:

File: src/bin/orc-export.rs (hunk 2 above)
Comment (klangner) [reply src/bin/orc-export.rs:91]: I will leave it as TODO. It can be added later.

File: src/bin/orc-export.rs (hunk 3 above)
Comment (klangner) [reply src/bin/orc-export.rs:20]: Yes, you right. Fixed

File: src/bin/orc-export.rs (hunk 1 above)
Comment (klangner) [reply src/bin/orc-export.rs:99]: Done

**Comment by Jefffrey** [comment]
Cheers :+1:

**Comment by klangner** [comment]
Thank you for your time too :-)

**Pull request merged**